node/pkg/driver/device_connectivity/device_connectivity_fc.go

```diff
@@ -0,0 +1,324 @@
+			errStrings = append(errStrings, e.Error())
+		}
+		devicePaths = append(devicePaths, dps...)
+	}
+
+	if len(devicePaths) == 0 && len(errStrings) != 0 {
+		err := errors.New(strings.Join(errStrings, ","))
+		return "", err
+	}
+
+	devicePathTosysfs := make(map[string]bool)
+	// Looping over the physical devices of the volume - /dev/sdX and store all the dm devices inside map.
+	for _, path := range devicePaths {
+		if path != "" { // since it may return empty items
+			mappedDevicePath, err := r.Helper.GetMultipathDisk(path)
+			if err != nil {
+				return "", err
+			}
+
+			if mappedDevicePath != "" {
+				devicePathTosysfs[mappedDevicePath] = true // map it in order to save uniq dm devices
+			}
+
+		}
+	}
+
+	var mps string
+	for key := range devicePathTosysfs {
+		mps += ", " + key
+	}
+	logger.Infof("GetMpathDevice: Found multipath devices: [%s] that relats to lunId=%d and arrayIdentifiers=%s", mps, lunId, arrayIdentifiers)
+
+	if len(devicePathTosysfs) > 1 {
+		return "", &MultipleDmDevicesError{volumeId, lunId, arrayIdentifiers, devicePathTosysfs}
+	}
+
+	var md string
+	for md = range devicePathTosysfs {
+		break // because its a single value in the map(1 mpath device, if not it should fail above), so just take the first
+	}
+	return md, nil
+}
+
+func (r OsDeviceConnectivityFc) FlushMultipathDevice(mpathDevice string) error {
+	return r.HelperScsiGeneric.FlushMultipathDevice(mpathDevice)
+}
+
+func (r OsDeviceConnectivityFc) RemovePhysicalDevice(sysDevices []string) error {
+	return r.HelperScsiGeneric.RemovePhysicalDevice(sysDevices)
+}
+
+// ============== OsDeviceConnectivityHelperFcInterface ==========================
+
+//go:generate mockgen -destination=../../../mocks/mock_OsDeviceConnectivityHelperFcInterface.go -package=mocks github.com/ibm/ibm-block-csi-driver/node/pkg/driver/device_connectivity OsDeviceConnectivityHelperFcInterface
+
+type OsDeviceConnectivityHelperFcInterface interface {
+	/*
+		This is helper interface for OsDeviceConnectivityFc.
+		Mainly for writting clean unit testing, so we can Mock this interface in order to unit test OsDeviceConnectivityFc logic.
+	*/
+	WaitForPathToExist(devicePath string, maxRetries int, intervalSeconds int) ([]string, bool, error)
+	GetMultipathDisk(path string) (string, error)
+	GetFcHostIDs() ([]int, error)
+}
+
+type OsDeviceConnectivityHelperFc struct {
+	executer executer.ExecuterInterface
+}
+
+const (
+	FC_HOST_SYSFS_PATH = "/sys/class/fc_host"
+)
+
+func NewOsDeviceConnectivityHelperFc(executer executer.ExecuterInterface) OsDeviceConnectivityHelperFcInterface {
+	return &OsDeviceConnectivityHelperFc{executer: executer}
+}
+
+func (o OsDeviceConnectivityHelperFc) GetFcHostIDs() ([]int, error) {
+	/*
+		Description:
+			This function find all the hosts IDs under directory /sys/class/fc_host/"
+			So the function goes over all the above hosts and return back only the host numbers as a list.
+	*/
+
+	targetNamePath := FC_HOST_SYSFS_PATH + "/host*/port_state"
```

Comment:
I think it is a port_state path, not the target name, please specify a reasonable name.

Reply:
Done